test/testcafe/spec/ChallengeAuthenticatorPhone_spec.js

```diff
@@ -0,0 +1,240 @@
+    const pageSubtitle = challengeFactorPageObject.getFormSubtitle();
+    await t.expect(challengeFactorPageObject.getSaveButtonLabel('input[type="submit"]')).eql('Verify');
+    await t.expect(pageSubtitle).contains('Calling');
+    await t.expect(pageSubtitle).contains('Enter the code below to verify.');
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-button-bar', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-fieldset', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.phone-authenticator-verify__button--primary', 'hide')).eql(true);
+    await t.expect(challengeFactorPageObject.elementExists('.phone-authenticator-verify__button--secondary')).eql(false);
+  });
+
+test
+  .requestHooks(invalidCodeMock)(`Entering invalid passcode results in an error`, async t => {
+    const challengeFactorPageObject = await setup(t);
+    await challengeFactorPageObject.clickElement('.phone-authenticator-verify__button--primary');
+    await challengeFactorPageObject.verifyFactor('credentials.passcode', 'abcd');
+    await challengeFactorPageObject.clickNextButton();
+    await challengeFactorPageObject.waitForErrorBox();
+    await t.expect(challengeFactorPageObject.getInvalidOTPError()).contains('Authentication failed');
+  });
+
+test
+  .requestHooks(logger, smsPrimaryMock)(`SMS Primary - Callout appears
+    after 30 seconds once the user clicks on send code button`, async t => {
+    const challengeFactorPageObject = await setup(t);
+    await challengeFactorPageObject.clickElement('.phone-authenticator-verify__button--primary');
+    await t.expect(challengeFactorPageObject.resendEmailView('.phone-authenticator-verify__resend-warning').hasClass('hide')).ok();
+    // wait for resend button to appear
+    await t.wait(32000);
+    // Making sure we keep polling while we wait for the resend view to appear
+    // Widget will poll with a refresh interval of 4000(comes from API).
+    // In 32000 seconds it will poll Math.floor(32000/4000) = 8 times
+    await t.expect(
+      logger.count(record => record.response.statusCode === 200
+        && record.request.url.match(/poll/))
+    ).eql(8);
+    await t.expect(challengeFactorPageObject.resendEmailView('.phone-authenticator-verify__resend-warning').hasClass('hide')).notOk();
+    const resendEmailView = challengeFactorPageObject.resendEmailView('.phone-authenticator-verify__resend-warning');
+    await t.expect(resendEmailView.innerText).eql('Haven\'t received an SMS? Send again');
+    await challengeFactorPageObject.clickSendAgainLink('.phone-authenticator-verify__resend-warning');
+    
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+      record.request.url.match(/poll|resend/)
+    )).eql(9);
+  });
+
+test
+  .requestHooks(logger, voicePrimaryMock)(`Voice Primary - Callout appears after 30 seconds once the user clicks on
```

Comment:
This is essentially the same as the test case above, but voice call is the primary option here. This takes 32 seconds to run, hence was wondering if we hang on to this or remove it?